code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -979,12 +979,18 @@
 	proc/cooltime()
 		if (src.warm)
 			spawn(4200)
-				src.warm = 0
-				for(var/reagent in heated_reagents)
-					src.reagents.del_reagent(reagent)
-				src.name = initial(name)
+				if(src)

```

Comment:
if (src) check added here. I don't know if that's smart. But i ran into a runtime error if the donk pocket was deleted between being warmed and cooling.